### PR TITLE
add vcpu related metrics within prometheus

### PR DESCRIPTION
**Pull request by vermavis**

### CODE REVIEW

File: vmtop.py [hunk 1]
```diff
@@ -1373,6 +1407,31 @@ def main():
...
             signal.SIGTERM: exit_gracefully
         }):
+            # Daemonizing kills all the previously opened sockets, so
+            # run prometheus after daemonizing
+            if args.prometheus:
+                try:
+                    from prometheus_client import start_http_server
+                    import socket
+                    host, port = args.prometheus.split(':')
+                    start_http_server(int(port), host)
```
Comment (jdesfossez) [inline vmtop.py:1417]: do we really need to do that multiple times ?
I think once in the main is enough.

Comment (pmmccorm) [reply vmtop.py]: I think all of the "try to import and possibly throw exception" code flow is a anti-pattern. All the imports should be at the top of the file. If something can't be imported set a global variable/flag and check that.

Comment (jdesfossez) [reply vmtop.py]: yes I agree with that

Comment (jdesfossez) [reply vmtop.py]: I will merge this PR because it's unrelated to that issue, and we'll make another PR to clean that up

**Pull request merged**